corehq/apps/integration/payments/services.py

```diff
@@ -1,21 +1,106 @@
+import uuid
+from dataclasses import asdict
 from datetime import datetime
 
+from django.utils.translation import gettext as _
+
 from dimagi.utils.chunked import chunked
-from corehq.apps.users.models import WebUser
+
 from corehq.apps.hqcase.api.updates import handle_case_update
+from corehq.apps.hqcase.utils import bulk_update_cases
+from corehq.apps.integration.payments.const import (
+    PAYMENT_SUCCESS_STATUS_CODE,
+    PaymentProperties,
+)
+from corehq.apps.integration.payments.exceptions import PaymentRequestError
+from corehq.apps.integration.payments.models import MoMoConfig
+from corehq.apps.integration.payments.schemas import (
+    PartyDetails,
+    PaymentTransferDetails,
+)
+from corehq.apps.users.models import WebUser
+from corehq.form_processor.models import CommCareCase
 
 CHUNK_SIZE = 100
 
 
+def request_payments_for_cases(case_ids, config):
+    payment_updates = []
+
+    for payment_case in CommCareCase.objects.get_cases(case_ids=case_ids):
```

Comment:
nit: probably not needed right now. I wonder if the number of case_ids could get a lot in which case this could be chunked in small groups.

Reply:
See latest commits